apps/website/src/pages/index.tsx

```diff
@@ -140,7 +144,80 @@ const getTwitchEmbed = (
   return url.toString();
 };
 
-const Home: NextPage = () => {
+// Define the schema for the expected structure of the feed entries
+const VideoSchema = z.object({
+  videoId: z.string(),
+  title: z.string(),
+  published: z.date(),
+});
+
+type Video = z.infer<typeof VideoSchema>;
+
+// Define the schema for the entire feed
+const FeedSchema = z.object({
+  feed: z.object({
+    entry: z.array(
+      z.object({
+        "yt:videoId": z.array(z.string()).nonempty(),
+        title: z.array(z.string()).nonempty(),
+        published: z.array(z.string()).nonempty(),
+      }),
+    ),
+  }),
+});
+
+type Props = {
+  videos: Video[];
+};
+
+const fetchYouTubeFeed = async (channelId: string): Promise<Video[]> => {
+  const url = `https://www.youtube.com/feeds/videos.xml?channel_id=${channelId}`;
+  const response = await axios.get(url);
+  const json = await parseStringPromise(response.data);
+  const parsedFeed = FeedSchema.parse(json);
+
+  return parsedFeed.feed.entry
+    .map((entry) => ({
+      videoId: entry["yt:videoId"][0],
+      title: entry.title[0],
+      published: new Date(entry.published[0]),
+    }))
+    .filter((video) => !video.title.includes("#shorts")) // exclude shorts
+    .map((video) => VideoSchema.parse(video)); // Validate each video against the schema
+};
+
+export const getStaticProps: GetStaticProps = async () => {
+  try {
+    const channelIds = ["UCbJ-1yM55NHrR1GS9hhPuvg", "UCfisf6HxiQr8_4mctNBm9cQ"];
+    const fetchPromises = channelIds.map((channelId) =>
+      fetchYouTubeFeed(channelId),
+    );
+    const videosArrays = await Promise.all(fetchPromises);
+
+    // Combine videos from all channels and sort by published date
+    const combinedVideos = videosArrays
+      .flat()
+      .sort((a, b) => b.published.getTime() - a.published.getTime());
+    const latestVideos = combinedVideos.slice(0, 4);
+
+    return {
+      props: {
+        videos: latestVideos,
+      },
+      revalidate: 1800, // revalidate after 30 minutes
+    };
+  } catch (error) {
+    return {
+      props: {
+        videos: [],
+        error: "Failed to fetch YouTube feed",
+      },
+      revalidate: 300, // revalidate after 5 mins
+    };
+  }
```

Comment:
Do we need to try/catch here? I believe we should actually just allow the error to be thrown which will allow Next to keep reusing the old data until it next attempts to revalidate?